.github/workflows/integration_tests.yml

```diff
@@ -37,9 +37,17 @@ jobs:
     runs-on: ${{ needs.start-runner.outputs.label }}
     timeout-minutes: 600
     steps:
+      - name: Determine branch ref
+        id: branch_ref
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "branch_ref=refs/heads/${{ github.head_ref }}" >> $GITHUB_OUTPUT
+          else
+            echo "branch_ref=${{ github.ref }}" >> $GITHUB_OUTPUT
```

Comment:
The else part is the easy part: for everything other than a pull request trigger, `github.ref `automatically evaluates to `refs/heads/<name of the branch>`.  We can save this to `$GITHUB_OUTPUT` environment variable to be used later.